src/components/multiselect/Multiselect.tsx

```diff
@@ -0,0 +1,252 @@
+      selectedItem: null,
+      items: allOptions,
+      stateReducer: (state, actionAndChanges) => {
+        const { changes, type } = actionAndChanges;
+        switch (type) {
+          case useSelect.stateChangeTypes.MenuKeyDownEnter:
+          case useSelect.stateChangeTypes.MenuKeyDownSpaceButton:
+          case useSelect.stateChangeTypes.ItemClick:
+            return {
+              ...changes,
+              isOpen: true // keep the menu open after selection.
+            };
+          default:
+            break;
+        }
+        return changes;
+      },
+      onStateChange: ({ type, selectedItem: newSelectedItem }) => {
+        switch (type) {
+          case useSelect.stateChangeTypes.MenuKeyDownSpaceButton:
+          case useSelect.stateChangeTypes.ItemClick:
+          case useSelect.stateChangeTypes.MenuKeyDownEnter:
+            // If new selection isn't already selected, we add it
+            if (
+              newSelectedItem &&
+              !selectedItems.find(
+                (item) => item.value === newSelectedItem.value
+              )
+            ) {
+              setSelectedItems(
+                addNewSelectedItem(
+                  selectedItems,
+                  newSelectedItem,
+                  allOptions.length
+                )
+              );
+              return;
+            }
+            // If new selection is already selected, we deselect it
+            if (
+              newSelectedItem &&
+              selectedItems.find((item) => item.value === newSelectedItem.value)
+            ) {
+              // Unless it's the only selected item
+              if (selectedItems.length === 1) {
+                return;
+              }
+              const newSelectedItems = selectedItems.filter((item) => {
+                return item.value !== newSelectedItem.value;
+              });
+              setSelectedItems(newSelectedItems);
+            }
+            break;
+          default:
+            break;
+        }
+      }
+    });
+
+  return (
+    <>
+      {caption && <div className="multiselect__caption">{caption}</div>}
+      <div className="multiselect" ref={ref} data-cy={dataCy}>
+        {/* eslint-disable react/jsx-props-no-spreading */}
+        {/* The downshift combobox works this way by design */}
+        <button
+          type="button"
+          className="multiselect focus-styling"
+          {...getToggleButtonProps(
+            getDropdownProps({ preventKeyAction: isDropdownOpen })
+          )}
+          onClick={() => {
+            setIsDropdownOpen(!isDropdownOpen);
+          }}
+          onKeyUp={(e) => {
+            if (e.key === "Enter") setIsDropdownOpen(!isDropdownOpen);
+          }}
+        >
+          <div className="multiselect__selected">
+            {selectedItems.map((singularitem, index) => {
+              const getSpace = index > 0 ? ", " : "";
+              return getSpace + t(singularitem.item);
+            })}
```

Comment:
Isn't this possible through the native `split()`function?

Reply:
doesn't split() just split a string? We are working with an an array of objects here.

Reply:
Sorry I meant `join` but I see your point with array of objects 👍